### PR TITLE
Add option not to retry updates after Get for sliding expiration

### PR DESCRIPTION
**Pull request by rhofstra-sefi**

### CODE REVIEW

**Comment by rhofstra-sefi** [comment]
@ealsur Let me know if you prefer me to update the changelog and package information or do it yourself

File: src/CosmosCache.cs [hunk 1]
```diff
@@ -115,8 +115,11 @@ public async Task<byte[]> GetAsync(string key, CancellationToken token = default
                 }
                 catch (CosmosException cosmosException) when (cosmosException.StatusCode == HttpStatusCode.PreconditionFailed)
                 {
-                    // Race condition on replace, we need to get the latest version of the item
-                    return await this.GetAsync(key, token).ConfigureAwait(false);
+                    if (this.options.RetrySlidingExpirationUpdates)
```
Comment (ealsur) [inline src/CosmosCache.cs:118]: It should be possible to cover this with unit tests (in the UT project) by mocking the client to return a 412 on the write operation and making sure the mock is invoked once

**Comment by rhofstra-sefi** [comment]
@ealsur Added unit tests.

**Comment by ealsur** [comment]
@rhofstra-sefi Do you need this to be released in a particular time frame?

**Comment by rhofstra-sefi** [comment]
> 
> 
> @rhofstra-sefi Do you need this to be released in a particular time frame?

@ealsur Sooner is better, sometime this week if possible?  No huge rush though, let me know and I can work with it.

**Comment by ealsur** [comment]
@rhofstra-sefi Released as preview4

**Pull request merged**